lib/PerlIO.pm

```diff
@@ -207,95 +267,101 @@ layer. Writes also behave like the C<:perlio> layer, as C<mmap()> for write
 needs extra house-keeping (to extend the file) which negates any advantage.
 
 The C<:mmap> layer will not exist if the platform does not support C<mmap()>.
+See L<PerlIO::mmap> for more information.
 
 =item :via
 
-Use C<:via(MODULE)> either in open() or binmode() to install a layer
-that does whatever transformation (for example compression /
-decompression, encryption / decryption) to the filehandle.
+C<:via(MODULE)> allows a transformation to be applied by an arbitrary Perl
+module, for example compression / decompression, encryption / decryption.
 See L<PerlIO::via> for more information.
 
+=item :scalar
+
+A layer implementing "in memory" files using scalar variables,
+automatically used in place of the platform defaults for IO when opening
+such a handle.  As such, the scalar is expected to act like a file, only
+containing or storing bytes.  See L<PerlIO::scalar> for more information.
+
 =back
 
 =head2 Alternatives to raw
 
 To get a binary stream an alternate method is to use:
 
-    open($fh,"whatever")
-    binmode($fh);
+    open(my $fh,"<","whatever") or die "open failed: $!";
+    binmode($fh) or die "open failed: $!";
 
-this has the advantage of being backward compatible with how such things have
-had to be coded on some platforms for years.
+This has the advantage of being backward compatible with older versions
+of Perl that did not use PerlIO or where C<:raw> was buggy (as it was
+before Perl 5.14).
 
 To get an unbuffered stream specify an unbuffered layer (e.g. C<:unix>)
 in the open call:
 
-    open($fh,"<:unix",$path)
+    open(my $fh,"<:unix",$path) or die "open failed: $!";
 
 =head2 Defaults and how to override them
 
 If the platform is MS-DOS like and normally does CRLF to "\n"
 translation for text files then the default layers are :
 
-  unix crlf
+  :unix:crlf
 
-(The low level "unix" layer may be replaced by a platform specific low
+(The low level C<:unix> layer may be replaced by a platform specific low
```

Comment:
I think this can safely be left out, as :win32 is known to be buggy, and no other platform specific layer exists.

Reply:
Oh is that what this is referring to? I wasn't sure what to do with :win32 as a whole.